pythonforandroid/archs.py

```diff
@@ -186,63 +247,61 @@ class ArchARM(Arch):
     @property
     def target(self):
         target_data = self.command_prefix.split('-')
-        return '-'.join(
-            ['armv7a', 'none', target_data[1], target_data[2]])
+        return '{triplet}{ndk_api}'.format(
+            triplet='-'.join(['armv7a', target_data[1], target_data[2]]),
+            ndk_api=self.ctx.ndk_api,
+        )
 
 
 class ArchARMv7_a(ArchARM):
     arch = 'armeabi-v7a'
-
-    def get_env(self, with_flags_in_cc=True, clang=False):
-        env = super(ArchARMv7_a, self).get_env(with_flags_in_cc, clang=clang)
-        env['CFLAGS'] = (env['CFLAGS'] +
-                         (' -march=armv7-a -mfloat-abi=softfp '
-                          '-mfpu=vfp -mthumb'))
-        env['CXXFLAGS'] = env['CFLAGS']
-        return env
+    arch_cflags = [
+        '-march=armv7-a',
+        '-mfloat-abi=softfp',
+        '-mfpu=vfp',
+        '-mthumb',
+    ]
 
 
 class Archx86(Arch):
     arch = 'x86'
     toolchain_prefix = 'x86'
     command_prefix = 'i686-linux-android'
     platform_dir = 'arch-x86'
-
-    def get_env(self, with_flags_in_cc=True, clang=False):
-        env = super(Archx86, self).get_env(with_flags_in_cc, clang=clang)
-        env['CFLAGS'] = (env['CFLAGS'] +
-                         ' -march=i686 -mtune=intel -mssse3 -mfpmath=sse -m32')
-        env['CXXFLAGS'] = env['CFLAGS']
-        return env
+    arch_cflags = [
+        '-march=i686',
+        '-mtune=intel',
+        '-mssse3',
+        '-mfpmath=sse',
+        '-m32',
+    ]
 
 
 class Archx86_64(Arch):
     arch = 'x86_64'
     toolchain_prefix = 'x86_64'
     command_prefix = 'x86_64-linux-android'
     platform_dir = 'arch-x86_64'
-
-    def get_env(self, with_flags_in_cc=True, clang=False):
-        env = super(Archx86_64, self).get_env(with_flags_in_cc, clang=clang)
-        env['CFLAGS'] = (env['CFLAGS'] +
-                         ' -march=x86-64 -msse4.2 -mpopcnt -m64 -mtune=intel')
-        env['CXXFLAGS'] = env['CFLAGS']
-        return env
+    arch_cflags = [
+        '-march=x86-64',
+        '-msse4.2',
+        '-mpopcnt',
+        '-m64',
+        '-mtune=intel',
+    ]
 
 
 class ArchAarch_64(Arch):
     arch = 'arm64-v8a'
     toolchain_prefix = 'aarch64-linux-android'
     command_prefix = 'aarch64-linux-android'
     platform_dir = 'arch-arm64'
-
-    def get_env(self, with_flags_in_cc=True, clang=False):
-        env = super(ArchAarch_64, self).get_env(with_flags_in_cc, clang=clang)
-        incpath = ' -I' + join(dirname(__file__), 'includes', 'arm64-v8a')
-        env['EXTRA_CFLAGS'] = incpath
-        env['CFLAGS'] += incpath
-        env['CXXFLAGS'] += incpath
-        if with_flags_in_cc:
-            env['CC'] += incpath
-            env['CXX'] += incpath
-        return env
+    arch_cflags = [
+        '-march=armv8-a',
+        # '-I' + join(dirname(__file__), 'includes', 'arm64-v8a'),
```

Comment:
This probably should be answered by @inclement, but maybe others could have the answer...

I'm not sure if we still need those commented lines I left in here...
Those includes were added around 2016 while adding Sdl2 bootstrap (b455e333), I modified the original lines to match the current situation and I commented them because it seems that I can build and run successfully the `testapp sqlite_openssl` with python 3 and aarch64...

Do you remember why we need this include?
Should we uncomment or remove those lines, alongside the target file?

Reply:
Sorry for the delay replying here, these lines probably had some meaning but will only have been debug comments, removing them is fine. Thanks for pointing it out.